meson.build

```diff
@@ -1,4 +1,7 @@
-project('ksmbsd-tools', 'c', version: '3.3.2', default_options: 'c_std=gnu99')
+project('ksmbsd-tools', 'c',
+  version: run_command(find_program('awk'), '''/define KSMBD_TOOLS_VERSION / \
+    { gsub(/"/,"",$3); printf "%s", $3 }''', 'include/version.h', check: true).stdout(),
+  default_options: 'c_std=gnu99')
```

Comment:
On an unrelated note, since the kernel recently switched to c11, would it make sense to do the same here? @namjaejeon

Reply:
@neheb Ah, Probably yes:) Could you please make a patch for this ?

Reply:
@atheik in meson's wrapdb, it's more common to use python scripts instead of sh or awk. Would that make sense here?

Reply:
@neheb,

Dependency to python is much bigger than dependency to some awk implementation. Note that this is just as easily done using sed only, but I chose to use awk since people tend to be unfamiliar with sed outside of its basic functionality.

Reply:
It's common specifically, and solely, because sh and awk are not (easily) available on Windows, while python is (and is already a dependency of Meson itself).

The flip side is that sh and awk are a lot faster than python, and can typically complete almost instantaneously, while python's interpreter startup cost means exiting immediately can take a whole tenth of a second even on a warm cache, and over a second on a cold one.

This is usually considered a worthwhile tradeoff in favor of python, if your goal is to be cross-platform, but I don't think that's the case here as ksmbd itself is an SMBD inside the (Linux) K(ernel).

Reply:
@eli-schwartz, @neheb,

> It's common specifically, and solely, because sh and awk are not (easily) available on Windows, while python is (and is already a dependency of Meson itself).

I hadn't considered this from that viewpoint. As for using awk, to me, there is value also in keeping the version extraction the same for both the autotools build and the meson build of ksmbd-tools.

Reply:
Speaking of muon:

```
/home/mangix/devstuff/ksmbd-tools/meson.build:5:5: error command failed: 'awk: cmd. line:1: /define KSMBD_TOOLS_VERSION / { gsub(/"/,"",$3); printf "%s", $3 } include/version.h
awk: cmd. line:1:                                                                                   ^ syntax error
'
  5 |     '/bin/sh',
          ^
/home/mangix/devstuff/ksmbd-tools/meson.build:4:12: error in function run_command()
  4 |   version: run_command(
                 ^
/home/mangix/devstuff/ksmbd-tools/meson.build:1:1: error in function project()
  1 | project(
      ^
```

wonder why.

Reply:
Fixed with

```
--- a/meson.build
+++ b/meson.build
@@ -4,9 +4,7 @@ project(
   version: run_command(
     '/bin/sh',
     '-c',
-    '''
-exec awk '/define KSMBD_TOOLS_VERSION / { gsub(/"/,"",$3); printf "%s", $3 }' include/version.h
-    ''',
+    'exec awk \'/define KSMBD_TOOLS_VERSION / { gsub(/"/,"",$3); printf "%s", $3 }\' include/version.h',
     check: true,
   ).stdout(),
   default_options: 'c_std=gnu99',
```

curious what the point of /bin/sh here is though.

Reply:
The reason for it is described at 643b71a073a5efb50343a87c8d114106f10aa954 and I figured out the same workaround, but, I believe this is a muon bug.